android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java

```diff
@@ -2,6 +2,50 @@
 
 public class Transaction {
 
+  private final String uid;
+  private final String hash;
+  private final String orderReference;
+  private final String status;
+  private String gateway;
+
+  public Transaction(String uid, String hash, String orderReference, String status,
+      String gateway) {
+
+    this.uid = uid;
+    this.hash = hash;
+    this.orderReference = orderReference;
+    this.status = status;
+    this.gateway = gateway;
+  }
+
+  public Transaction() {
```

Comment:
Did you consider creating a static method to create a `FailedTransaction` instead of a constructor with no parameters?

Reply:
So that's how you create an empty constructor with a verbose name. But FailedTransaction is not the best name. ErrorTransaction is a better suited name I think

Reply:
Agree, i just wanted to explain the idea